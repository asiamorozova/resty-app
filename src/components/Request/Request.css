.Request{
    input{
        display: block;
        margin: 1rem 0;
      
    }

    label{
        padding: 0.6rem;
        border-radius: 9px;
        margin: .6rem 0.3rem;
        cursor: wait;


    }
    input[type='radio']{
        display: none;
    }

    input[type="radio"]:checked + label {
        background-color: aliceblue;
        color: lightgray;
    }

    textarea {
        display: block;
        margin: 2rem;
    }
    button {
        display: block;
        border: none;
        display: inline;
        color: whitesmoke;
    }
}