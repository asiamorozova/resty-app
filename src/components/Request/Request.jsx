import React from 'react';
import PropTypes from 'prop-types';

const Request = ({ url, onUrlChange, body, onBodyChange, onMethodChange, onSubmit }) => (
  <form nonSubmit={onSubmit} >
    <input type="text" value={url} onChange={onUrlChange} />
    <label>
            POST:
      <input type="radio" name="method" value="POST" onChange={onMethodChange} />
    </label>
    <label>
            GET:
      <input type="radio" name="method" value="GET" onChange={onMethodChange} />
    </label>
    <label>
            PATCH:
      <input type="radio" name="method" value="PATCH" onChange={onMethodChange} />
    </label>
    <label>
            PUT:
      <input type="radio" name="method" value="PUT" onChange={onMethodChange} />
    </label>
    <label>
            DELETE:
      <input type="radio" name="method" value="DELETE" onChange={onMethodChange} />
    </label>
    <textarea value ={body} onChange={onBodyChange}></textarea>
    <button>Submit</button>
  </form>
);

Request.propTypes = {
  url: PropTypes.string.isRequired,
  method: PropTypes.string.isRequired,
  body: PropTypes.object.isRequired,
  onUrlChange: PropTypes.func.isRequired,
  onMethodChange: PropTypes.func.isRequired, 
  onBodyChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired
};

export default Request; 
