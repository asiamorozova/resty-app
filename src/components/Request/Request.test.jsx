import React from 'react';
import { shallow } from 'enzyme';
import Request from './Request';

describe('Request concomponent', () => {
  it('matches a snapshot of Request.jsx', () => {
    const wrapper = shallow(<Request
      url='test url'
      method='GET'
      textBody='JSON body'
      onChange={()=>{}}
      onSubmit={()=>{}}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
