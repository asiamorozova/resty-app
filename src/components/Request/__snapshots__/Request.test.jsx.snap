// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Request concomponent matches a snapshot of Request.jsx 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <label>
       URL: 
      <input
        name="url"
        onChange={[Function]}
        type="text"
        value="test url"
      />
    </label>
    <div>
      <label>
         
        <input
          name="method"
          type="radio"
          value="get"
          vonChange={[Function]}
        />
         GET
      </label>
      <label>
        <input
          name="method"
          type="radio"
          value="post"
          vonChange={[Function]}
        />
         POST
      </label>
      <label>
        <input
          name="method"
          onChange={[Function]}
          type="radio"
          value="put"
        />
         PUT
      </label>
      <label>
        <input
          name="method"
          onChange={[Function]}
          type="radio"
          value="patch"
        />
        PATCH
      </label>
      <label>
        <input
          name="method"
          onChange={[Function]}
          type="radio"
          value="delete"
        />
         DELETE
      </label>
    </div>
    <textarea
      name="textBody"
      onChange={[Function]}
      placeholder="JSON Body"
      value="JSON body"
    />
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
